metricbeat/module/golang/expvar/expvar.go

```diff
@@ -0,0 +1,82 @@
+package expvar
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/metricbeat/helper"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/mb/parse"
+	"github.com/elastic/beats/metricbeat/module/golang"
+)
+
+const (
+	defaultScheme = "http"
+	defaultPath   = "/debug/vars"
+)
+
+var (
+	hostParser = parse.URLHostParserBuilder{
+		DefaultScheme: defaultScheme,
+		DefaultPath:   defaultPath,
+		PathConfigKey: "expvar.path",
+	}.Build()
+)
+
+// init registers the MetricSet with the central registry.
+// The New method will be called after the setup of the module and before starting to fetch data
+func init() {
+	if err := mb.Registry.AddMetricSet("golang", "expvar", New, hostParser); err != nil {
+		panic(err)
+	}
+}
+
+// MetricSet type defines all fields of the MetricSet
+// As a minimum it must inherit the mb.BaseMetricSet fields, but can be extended with
+// additional entries. These variables can be used to persist data or configuration between
+// multiple fetch calls.
+type MetricSet struct {
+	mb.BaseMetricSet
+	http *helper.HTTP
+}
+
+// New create a new instance of the MetricSet
+// Part of new is also setting up the configuration by processing additional
+// configuration entries if needed.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+
+	logp.Warn("EXPERIMENTAL: The golang expvar metricset is experimental")
+
+	config := struct{}{}
+
+	if err := base.Module().UnpackConfig(&config); err != nil {
+		return nil, err
+	}
+
+	return &MetricSet{
+		BaseMetricSet: base,
+		http:          helper.NewHTTP(base),
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right format
+// It returns the event which is then forward to the output. In case of an error, a
+// descriptive error must be returned.
+func (m *MetricSet) Fetch() (common.MapStr, error) {
+
+	json, err := m.http.FetchJSON()
+
+	if err != nil {
+		return nil, err
+	}
+
+	event := common.MapStr{}
+
+	//export all expvar keys, except memstats
+	json["memstats"] = nil
+	event = json
+
+	//flatten cmdline
+	event["cmdline"] = golang.GetCmdStr(json["cmdline"])
+
```

Comment:
We need to set `_namespace` here to make it possible to have its own namespace for expvar. See https://github.com/elastic/beats/blob/master/metricbeat/module/prometheus/collector/collector.go#L91 as an example. Namespace config will also be needed.